aips/aip-x.md

```diff
@@ -0,0 +1,707 @@
+| plainText             | string          | ✓         | The constructed message using the standardized message format.                                                                        |
+| signature             | bytes           | ✓         | The signature of the message. This must conform to the domain separated signing algorithm.                                            |
+| type                  | string          | ✓         | The public key scheme used to sign the message (e.g. 'ed25519', 'single_key', 'multi_key', 'multi_ed25519', etc.)                     |
+
+### AptosSignInOutput
+
+When the user has approved and signed the message, the wallet will respond with an `AptosSignInOutput` to the application. The application should then use this response to verify if the outputs are valid with the inputs.
+
+| Name              | Type          | Mandatory | Description                                                                                                                        |
+| ----------------- | ------------- | --------- | ---------------------------------------------------------------------------------------------------------------------------------- |
+| account.address   | string        | ✓         | The address of the user that is signing in.                                                                                        |
+| account.publicKey | PublicKey     | ✓         | The public key of the user that is signing in.                                                                                     |
+| message           | SignInMessage | ✓         | A SignInMessage object that contains information about the message that was signed, it's signature, and plain text representation. |
+| signature         | Signature     | ✓         | The signature of the message.                                                                                                      |
+
+### Message Construction
+
+The proposed plaintext representation will be an adaptation of the standards specified from [EIP-4361](https://eips.ethereum.org/EIPS/eip-4361) and [CAIP-122](https://chainagnostic.org/CAIPs/caip-122). If fields are not present, omit them from the plaintext message.
+
+#### Message Format
+
+```
+${domain} wants you to sign in with your Aptos account:
+${address}
+
+${statement}
+
+URI: ${uri}
+Version: ${version}
+Chain ID: ${chain-id}
+Nonce: ${nonce}
+Issued At: ${issued-at}
+Expiration Time: ${expiration-time}
+Not Before: ${not-before}
+Request ID: ${request-id}
+Resources:
+- ${resources[0]}
+- ${resources[1]}
+...
+- ${resources[n]}
+```
+
+#### Example Messages
+
+_Minimal Example_
+
+```
+example.com wants you to sign in with your Aptos account:
+0x0000000000000000000000000000000000000000000000000000000000000001
+
+URI: https://example.com
+Version: 1
+Chain ID: mainnet
+```
+
+_Detailed Example_
```

Comment:
It would be very helpful if this AIP described when field like `nonce` should be set in the message. 

This section on examples could be a good place to do that: go over a few sample messages and detail an application where that message suffices. I am actually wondering where the "minimal example" above would actually arise.